### PR TITLE
[SDS-242] Optimized-flag not valid for hardware

### PR DESCRIPTION
**Pull request by QFer**
* Full state projection is never True for a created job that runs on hardware
* Fixed a mypy error

### CODE REVIEW

File: src/quantuminspire/api.py [hunk 1]
```diff
@@ -501,6 +501,7 @@ def _create_job(self, name: str, asset: Dict[str, Any], number_of_shots: int,
             The properties describing the new job.
             See :meth:`~.get_job` for a description of the job properties.
         """
+        full_state_projection = full_state_projection and not backend_type.get("is_hardware_backend", False)
```
Comment (oschusler) [inline src/quantuminspire/api.py:504]: Shouldn't the fallback value be `True`. If we don't know whether it is a hardware_backend or not (for whatever value), safest would be to default to see the algorithm as non-deterministic?

**Review by oschusler — CHANGES_REQUESTED** [review]
This branch is out of date

File: src/quantuminspire/api.py (hunk 1 above)
Comment (QFer) [reply src/quantuminspire/api.py:504]: That's exactly why the default is False, non-deterministic (full_state_projection = False). See also knowledge base, optimizations of simulation: "Full state projection is the keyword used internally in Quantum Inspire to do optimization using a single determination of the full final state. By default, this argument is set to False"

**Comment by QFer** [comment]
I'll rebase this change

File: src/quantuminspire/api.py (hunk 1 above)
Comment (QFer) [reply src/quantuminspire/api.py:504]: !! I see, you are right, I will change to True, I overlooked the not

**Review by oschusler — APPROVED** [review]

**Pull request merged**